tests/testsuite/check.rs

```diff
@@ -1024,3 +1024,59 @@ fn rustc_workspace_wrapper_excludes_published_deps() {
         .with_stdout_does_not_contain("WRAPPER CALLED: rustc --crate-name baz [..]")
         .run();
 }
+
+#[cargo_test]
+fn warn_manifest_package_and_project() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [project]
+                name = "foo"
+                version = "0.0.1"
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("check")
+        .with_status(0)
```

Comment:
nit: I believe we don't need this

https://github.com/rust-lang/cargo/blob/1a5a7ff321dbcb2586088b136ee7aafda18a9147/crates/cargo-test-support/src/lib.rs#L578-L581

```suggestion
```